Angular/src/app/admin-auth-guard/admin-auth-guard.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from "@angular/core";
+import { Router, CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from "@angular/router";
+import { CookieService } from 'ngx-cookie-service';
+import { decode, verify } from 'jsonwebtoken'
+import { AuthGuardComponent } from '../auth-guard/auth-guard.component';
+
+@Injectable()
+export class AdminAuthGuardComponent implements CanActivate {
+    constructor(private router: Router, private cookieService: CookieService, 
+      private authGuard: AuthGuardComponent){}
+
+    cookieName = 'adminMode'
+
+    isAdminMode(){
+      return this.cookieService.get(this.cookieName) == 'true'
+    }
```

Comment:
I would add return types to your methods, and types to your variables in this file.
E.g., `isAdminMode(): boolean`

Reply:
sorted